src/mmw/js/src/core/models.js

```diff
@@ -290,46 +289,44 @@ var ObservationsLayerGroupModel = LayerGroupModel.extend({
         layers: null,
     },
 
-    fetchLayers: function(map) {
+    fetchLayers: function() {
         var self = this,
-            pointSrcAPIUrl = '/mmw/modeling/point-source/';
-        var vizer = new VizerLayers();
-        this.set('polling', true);
+            weatherStationAPIUrl = '/mmw/modeling/weather-stations/';
 
-        $.when(vizer.getLayers(), $.ajax({ 'url': pointSrcAPIUrl, 'type': 'GET'}))
-            .done(function(observationLayers, pointSourceData) {
-                self.set({
-                    'polling': false,
-                    'error': null,
-                });
+        if(self.get('polling')) {
+            // Avoid calling fetchLayers again while polling.
+            return;
+        }
 
-                var observationLayerObjects =_.map(observationLayers, function(leafletLayer, display) {
-                        return {
-                                leafletLayer: leafletLayer,
-                                display: display,
-                                active: false,
-                                layerType: 'observations'
-                            };
-                    }),
-                    observationLayersCollection = new Backbone.Collection(observationLayerObjects);
+        this.set('polling', true);
+
+        $.when($.ajax({ 'url': weatherStationAPIUrl, 'type': 'GET'}))
```

Comment:
We don't  need to wrap this in `$.when`. Since `$.ajax` returns a promise, we can call `$.ajax( ... ).done( ... )` directly.